node/pkg/governor/governor.go

```diff
@@ -836,46 +913,46 @@ func computeValue(amount *big.Int, token *tokenEntry) (uint64, error) {
 	return value, nil
 }
 
-// TrimAndSumValueForChain calculates the `sum` of `Transfer`s for a given chain `emitter`. In effect, it represents a
+// TrimAndSumValueForChain calculates the `sum` of `Transfer`s for a given chain `chainEntry`. In effect, it represents a
 // chain's "Governor Usage" for a given 24 hour period.
 // This sum may be reduced by the sum of 'flow cancelling' transfers: that is, transfers of an allow-listed token
 // that have the `emitter` as their destination chain.
 // The resulting `sum` return value therefore represents the net flow across a chain when taking flow-cancelling tokens
 // into account. Therefore, this value should never be less than 0 and should never exceed the "Governor limit" for the chain.
-// As a side-effect, this function modifies the parameter `emitter`, updating its `transfers` field so that it only includes
+// As a side-effect, this function modifies the parameter `chainEntry`, updating its `transfers` field so that it only includes
 // filtered `Transfer`s (i.e. outgoing `Transfer`s newer than `startTime`).
+// Returns an error if the sum cannot be calculated. The transfers field will still be updated in this case. When
+// an error condition occurs, this function returns the chain's `dailyLimit` as the sum. This should result in the
+// chain appearing at maximum capacity from the perspective of the Governor, and therefore cause new transfers to be
+// queued until space opens up.
 // SECURITY Invariant: The `sum` return value should never be less than 0
-// SECURITY Invariant: The `sum` return value should never exceed the "Governor limit" for the chain
-func (gov *ChainGovernor) TrimAndSumValueForChain(emitter *chainEntry, startTime time.Time) (sum uint64, err error) {
-	// Sum the value of all outgoing transfers
-	var sumOutgoing int64
-	sumOutgoing, emitter.transfers, err = gov.TrimAndSumValue(emitter.transfers, startTime)
+func (gov *ChainGovernor) TrimAndSumValueForChain(chainEntry *chainEntry, startTime time.Time) (sum uint64, err error) {
+	if chainEntry == nil {
+		// We don't expect this to happen but this prevents a nil pointer deference
+		return 0, errors.New("TrimAndSumValeForChain parameter chainEntry must not be nil")
```

Comment:
Seems like a program bug, so maybe panic is better?

Reply:
Do you know the effects this would have on the processor? I want to avoid a situation where a bug in the Governor logic causes it or related processes to restart. I see your point that this is a serious problem we don't want to encounter but I'm hesitant to add panics or otherwise affect liveness

Reply:
Returning an error would cause the processor to restart, which is a silent failure unless someone is watching the logs.

A panic would kill the application, which should be much more visible.

My point was is that this should only happen in the case of a program bug, as in someone broke the code, which should be detected during testing, and never make it to production.

Reply:
The code that calls should `TrimAndSumValueForChain` log and swallows the error in all cases rather than propagating it up to the processor. (This is the intention anyway, let me know if you see otherwise.)

The idea here wasn't too fail silently or loudly in the sense of causing the processor to restart. Instead the error is logged.

My understanding is that the only path to restart the processor is via [returning an error from `CheckPending`](https://github.com/wormhole-foundation/wormhole/blob/80c26eb8729a4aab5ff338c1ac01a526531ead50/node/pkg/processor/processor.go#L264-L267). Errors from `ProcessMsg` do not cause a restart; [instead there's a continue.](https://github.com/wormhole-foundation/wormhole/blob/80c26eb8729a4aab5ff338c1ac01a526531ead50/node/pkg/processor/processor.go#L231)